src/sentry/static/sentry/app/views/organizationIntegrations/integrationIcon.tsx

```diff
@@ -12,16 +12,40 @@ type Props = {
 
 type IconProps = Pick<Props, 'size'>;
 
-const Icon = styled('img')<IconProps>`
+const StyledIcon = styled('img')<IconProps>`
   height: ${p => p.size}px;
   width: ${p => p.size}px;
   border-radius: 2px;
   display: block;
 `;
 
+class Icon extends React.Component<Props> {
+  state = {
+    imgSrc: '',
+  };
+
+  componentDidMount() {
+    const {integration} = this.props;
+    this.setState({imgSrc: integration.icon});
```

Comment:
@NisanthanNanthakumar Is this going to cause the component to mount with the wrong image for a moment? Probably not noticeable though.

Reply:
Well it will mount and then the img tag will do it's native fetch.